theano/tensor/nnet/opt.py

```diff
@@ -62,7 +62,7 @@ def local_inplace_sparse_block_outer(node):
 # Conv opts
 @local_optimizer([AbstractConv2d])
 def local_abstractconv_gemm(node):
-    if theano.config.cxx == "":
+    if theano.config.cxx == "" or theano.config.blas.ldflags:
         return
```

Comment:
Shouldn't this be `or not theano.config.blas.ldflags`? Otherwise you're skipping it exactly when BLAS is available.